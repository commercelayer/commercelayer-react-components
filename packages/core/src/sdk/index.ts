import sdk, { type CommerceLayerClient } from "@commercelayer/sdk"
import { jwtDecode } from "jwt-decode"
import type { Jwt, RequestConfig } from "src/types/index.js"

export function getSdk({ accessToken }: RequestConfig): CommerceLayerClient {
  const organization = jwtDecode<Jwt>(accessToken).organization.slug
  return sdk({ accessToken, organization })
}
