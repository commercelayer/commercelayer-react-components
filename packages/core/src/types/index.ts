export type { RequestConfig } from "./base"
