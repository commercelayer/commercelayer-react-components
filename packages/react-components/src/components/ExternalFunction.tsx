import ExternalFunctionContext, {
  callExternalFunction
} from '#context/ExternalFunctionContext'
import { type ReactNode } from 'react'

interface Props {
  children: ReactNode
  url: string
}

export function ExternalFunction({ children, url }: Props): JSX.Element {
  return (
    <ExternalFunctionContext.Provider value={{ url, callExternalFunction }}>
      {children}
    </ExternalFunctionContext.Provider>
  )
}

export default ExternalFunction
