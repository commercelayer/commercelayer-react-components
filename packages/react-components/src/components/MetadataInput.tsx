import BaseInput from '#components/utils/BaseInput'
import { type BaseInputComponentProps } from '#typings'

type Props = BaseInputComponentProps &
  JSX.IntrinsicElements['input'] &
  JSX.IntrinsicElements['textarea']

export function MetadataInput(props: Props): JSX.Element {
  return <BaseInput data-metadata {...props} />
}

export default MetadataInput
