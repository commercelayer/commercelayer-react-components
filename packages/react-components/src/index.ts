export * from '#components/errors/Errors'
export * from '#components/ExternalFunction'
export * from '#components/MetadataInput'
export * from '#components/SubmitButton'
export * from '#components/addresses/Address'
export * from '#components/addresses/AddressCountrySelector'
export * from '#components/addresses/AddressField'
export * from '#components/addresses/AddressInput'
export * from '#components/addresses/AddressStateSelector'
export * from '#components/addresses/AddressesContainer'
export * from '#components/addresses/AddressesEmpty'
export * from '#components/addresses/BillingAddressContainer'
export * from '#components/addresses/BillingAddressForm'
export * from '#components/addresses/SaveAddressesButton'
export * from '#components/addresses/ShippingAddressContainer'
export * from '#components/addresses/ShippingAddressForm'
export * from '#components/auth/CommerceLayer'
export * from '#components/customers/CustomerAddressForm'
export * from '#components/customers/CustomerContainer'
export * from '#components/customers/CustomerField'
export * from '#components/customers/CustomerInput'
export * from '#components/customers/SaveCustomerButton'
export * from '#components/gift_cards/GiftCard'
export * from '#components/gift_cards/GiftCardContainer'
export * from '#components/gift_cards/GiftCardCurrencySelector'
export * from '#components/gift_cards/GiftCardInput'
export * from '#components/gift_cards/GiftCardOrCouponCode'
export * from '#components/gift_cards/GiftCardOrCouponForm'
export * from '#components/gift_cards/GiftCardOrCouponInput'
export * from '#components/gift_cards/GiftCardOrCouponRemoveButton'
export * from '#components/gift_cards/GiftCardOrCouponSubmit'
export * from '#components/line_items/LineItemsContainer'
export * from '#components/line_items/LineItem'
export * from '#components/line_items/LineItemAmount'
export * from '#components/line_items/LineItemCode'
export * from '#components/line_items/LineItemField'
export * from '#components/line_items/LineItemImage'
export * from '#components/line_items/LineItemName'
export * from '#components/line_items/LineItemOption'
export * from '#components/line_items/LineItemOptions'
export * from '#components/line_items/LineItemQuantity'
export * from '#components/line_items/LineItemRemoveLink'
export * from '#components/line_items/LineItemsCount'
export * from '#components/line_items/LineItemsEmpty'
export * from '#components/orders/AddToCartButton'
export * from '#components/orders/AdjustmentAmount'
export * from '#components/orders/CartLink'
export * from '#components/orders/CheckoutLink'
export * from '#components/orders/DiscountAmount'
export * from '#components/orders/GiftCardAmount'
export * from '#components/orders/OrderContainer'
export * from '#components/orders/OrderList'
export * from '#components/orders/OrderListEmpty'
export * from '#components/orders/OrderListRow'
export * from '#components/orders/OrderNumber'
export * from '#components/orders/OrderStorage'
export * from '#components/orders/PaymentMethodAmount'
export * from '#components/orders/PlaceOrderButton'
export * from '#components/orders/PlaceOrderContainer'
export * from '#components/orders/PrivacyAndTermsCheckbox'
export * from '#components/orders/ShippingAmount'
export * from '#components/orders/SubTotalAmount'
export * from '#components/orders/TaxesAmount'
export * from '#components/orders/TotalAmount'
export * from '#components/parcels/ParcelField'
export * from '#components/parcels/ParcelLineItem'
export * from '#components/parcels/ParcelLineItemField'
export * from '#components/parcels/ParcelLineItemsCount'
export * from '#components/parcels/Parcels'
export * from '#components/parcels/ParcelsCount'
export * from '#components/payment_methods/PaymentMethod'
export * from '#components/payment_methods/PaymentMethodName'
export * from '#components/payment_methods/PaymentMethodPrice'
export * from '#components/payment_methods/PaymentMethodRadioButton'
export * from '#components/payment_methods/PaymentMethodsContainer'
export * from '#components/payment_source/PaymentSource'
export * from '#components/payment_source/PaymentSourceBrandIcon'
export * from '#components/payment_source/PaymentSourceBrandName'
export * from '#components/payment_source/PaymentSourceDetail'
export * from '#components/payment_source/PaymentSourceEditButton'
export * from '#components/prices/Price'
export * from '#components/prices/PricesContainer'
export * from '#components/shipments/Shipment'
export * from '#components/shipments/ShipmentField'
export * from '#components/shipments/ShipmentsContainer'
export * from '#components/shipments/ShipmentsCount'
export * from '#components/shipping_methods/ShippingMethod'
export * from '#components/shipping_methods/ShippingMethodName'
export * from '#components/shipping_methods/ShippingMethodPrice'
export * from '#components/shipping_methods/ShippingMethodRadioButton'
export * from '#components/skus/AvailabilityContainer'
export * from '#components/skus/AvailabilityTemplate'
export * from '#components/skus/DeliveryLeadTime'
export * from '#components/skus/SkuField'
export * from '#components/skus/SkuList'
export * from '#components/skus/SkuListsContainer'
export * from '#components/skus/Skus'
export * from '#components/skus/SkusContainer'
export * from '#components/stock_transfers/StockTransfer'
export * from '#components/stock_transfers/StockTransferField'
export * from '#typings/errors'
