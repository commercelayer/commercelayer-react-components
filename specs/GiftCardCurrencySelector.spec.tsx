import React from 'react'
import { GiftCardCurrencySelector } from '../src'
import renderer from 'react-test-renderer'
import components from '../src/config/components'
import currencyOptions from '../src/config/currency.json'

const propTypes = components.GiftCardCurrencySelector.propTypes

test('<GiftCardCurrencySelector/>', () => {
  expect.assertions(5)
  const component = renderer.create(<GiftCardCurrencySelector />)
  const tree = component.toJSON()
  const root = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(propTypes.children)
  expect(proptypes.placeholder).toBe(propTypes.placeholder)
  expect(proptypes.value).toBe(propTypes.value)
  expect(proptypes.required).toBe(propTypes.required)
})

test('<GiftCardCurrencySelector check children />', () => {
  const component = renderer.create(<GiftCardCurrencySelector />)
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const options = [...currencyOptions]
  const childrenRendered = rendered.rendered
  options.map(({ value, label }) => {
    const option = [
      {
        instance: null,
        nodeType: 'host',
        props: { value: value, children: label },
        rendered: [label],
        type: 'option',
      },
    ]
    expect(childrenRendered).toEqual(expect.arrayContaining(option))
  })
  expect(tree).toMatchSnapshot()
  expect(rendered.props.name).toBe('currencyCode')
  expect(rendered.props.required).toBeTruthy()
  expect(rendered.type).toStrictEqual('select')
})
