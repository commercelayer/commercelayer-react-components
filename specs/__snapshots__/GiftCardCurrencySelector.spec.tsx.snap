// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftCardCurrencySelector check children /> 1`] = `
<select
  defaultValue=""
  name="currencyCode"
  required={true}
>
  <option
    value=""
  >
    Select an option
  </option>
  <option
    value="AED"
  >
    AED
  </option>
  <option
    value="AFN"
  >
    AFN
  </option>
  <option
    value="ALL"
  >
    ALL
  </option>
  <option
    value="AMD"
  >
    AMD
  </option>
  <option
    value="ANG"
  >
    ANG
  </option>
  <option
    value="AOA"
  >
    AOA
  </option>
  <option
    value="ARS"
  >
    ARS
  </option>
  <option
    value="AUD"
  >
    AUD
  </option>
  <option
    value="AWG"
  >
    AWG
  </option>
  <option
    value="AZN"
  >
    AZN
  </option>
  <option
    value="BAM"
  >
    BAM
  </option>
  <option
    value="BBD"
  >
    BBD
  </option>
  <option
    value="BDT"
  >
    BDT
  </option>
  <option
    value="BGN"
  >
    BGN
  </option>
  <option
    value="BHD"
  >
    BHD
  </option>
  <option
    value="BIF"
  >
    BIF
  </option>
  <option
    value="BMD"
  >
    BMD
  </option>
  <option
    value="BND"
  >
    BND
  </option>
  <option
    value="BOB"
  >
    BOB
  </option>
  <option
    value="BRL"
  >
    BRL
  </option>
  <option
    value="BSD"
  >
    BSD
  </option>
  <option
    value="BTN"
  >
    BTN
  </option>
  <option
    value="BWP"
  >
    BWP
  </option>
  <option
    value="BYN"
  >
    BYN
  </option>
  <option
    value="BYR"
  >
    BYR
  </option>
  <option
    value="BZD"
  >
    BZD
  </option>
  <option
    value="CAD"
  >
    CAD
  </option>
  <option
    value="CDF"
  >
    CDF
  </option>
  <option
    value="CHF"
  >
    CHF
  </option>
  <option
    value="CLF"
  >
    CLF
  </option>
  <option
    value="CLP"
  >
    CLP
  </option>
  <option
    value="CNY"
  >
    CNY
  </option>
  <option
    value="COP"
  >
    COP
  </option>
  <option
    value="CRC"
  >
    CRC
  </option>
  <option
    value="CUC"
  >
    CUC
  </option>
  <option
    value="CUP"
  >
    CUP
  </option>
  <option
    value="CVE"
  >
    CVE
  </option>
  <option
    value="CZK"
  >
    CZK
  </option>
  <option
    value="DJF"
  >
    DJF
  </option>
  <option
    value="DKK"
  >
    DKK
  </option>
  <option
    value="DOP"
  >
    DOP
  </option>
  <option
    value="DZD"
  >
    DZD
  </option>
  <option
    value="EGP"
  >
    EGP
  </option>
  <option
    value="ERN"
  >
    ERN
  </option>
  <option
    value="ETB"
  >
    ETB
  </option>
  <option
    value="EUR"
  >
    EUR
  </option>
  <option
    value="FJD"
  >
    FJD
  </option>
  <option
    value="FKP"
  >
    FKP
  </option>
  <option
    value="GBP"
  >
    GBP
  </option>
  <option
    value="GEL"
  >
    GEL
  </option>
  <option
    value="GHS"
  >
    GHS
  </option>
  <option
    value="GIP"
  >
    GIP
  </option>
  <option
    value="GMD"
  >
    GMD
  </option>
  <option
    value="GNF"
  >
    GNF
  </option>
  <option
    value="GTQ"
  >
    GTQ
  </option>
  <option
    value="GYD"
  >
    GYD
  </option>
  <option
    value="HKD"
  >
    HKD
  </option>
  <option
    value="HNL"
  >
    HNL
  </option>
  <option
    value="HRK"
  >
    HRK
  </option>
  <option
    value="HTG"
  >
    HTG
  </option>
  <option
    value="HUF"
  >
    HUF
  </option>
  <option
    value="IDR"
  >
    IDR
  </option>
  <option
    value="ILS"
  >
    ILS
  </option>
  <option
    value="INR"
  >
    INR
  </option>
  <option
    value="IQD"
  >
    IQD
  </option>
  <option
    value="IRR"
  >
    IRR
  </option>
  <option
    value="ISK"
  >
    ISK
  </option>
  <option
    value="JMD"
  >
    JMD
  </option>
  <option
    value="JOD"
  >
    JOD
  </option>
  <option
    value="JPY"
  >
    JPY
  </option>
  <option
    value="KES"
  >
    KES
  </option>
  <option
    value="KGS"
  >
    KGS
  </option>
  <option
    value="KHR"
  >
    KHR
  </option>
  <option
    value="KMF"
  >
    KMF
  </option>
  <option
    value="KPW"
  >
    KPW
  </option>
  <option
    value="KRW"
  >
    KRW
  </option>
  <option
    value="KWD"
  >
    KWD
  </option>
  <option
    value="KYD"
  >
    KYD
  </option>
  <option
    value="KZT"
  >
    KZT
  </option>
  <option
    value="LAK"
  >
    LAK
  </option>
  <option
    value="LBP"
  >
    LBP
  </option>
  <option
    value="LKR"
  >
    LKR
  </option>
  <option
    value="LRD"
  >
    LRD
  </option>
  <option
    value="LSL"
  >
    LSL
  </option>
  <option
    value="LTL"
  >
    LTL
  </option>
  <option
    value="LVL"
  >
    LVL
  </option>
  <option
    value="LYD"
  >
    LYD
  </option>
  <option
    value="MAD"
  >
    MAD
  </option>
  <option
    value="MDL"
  >
    MDL
  </option>
  <option
    value="MGA"
  >
    MGA
  </option>
  <option
    value="MKD"
  >
    MKD
  </option>
  <option
    value="MMK"
  >
    MMK
  </option>
  <option
    value="MNT"
  >
    MNT
  </option>
  <option
    value="MOP"
  >
    MOP
  </option>
  <option
    value="MRO"
  >
    MRO
  </option>
  <option
    value="MUR"
  >
    MUR
  </option>
  <option
    value="MVR"
  >
    MVR
  </option>
  <option
    value="MWK"
  >
    MWK
  </option>
  <option
    value="MXN"
  >
    MXN
  </option>
  <option
    value="MYR"
  >
    MYR
  </option>
  <option
    value="MZN"
  >
    MZN
  </option>
  <option
    value="NAD"
  >
    NAD
  </option>
  <option
    value="NGN"
  >
    NGN
  </option>
  <option
    value="NIO"
  >
    NIO
  </option>
  <option
    value="NOK"
  >
    NOK
  </option>
  <option
    value="NPR"
  >
    NPR
  </option>
  <option
    value="NZD"
  >
    NZD
  </option>
  <option
    value="OMR"
  >
    OMR
  </option>
  <option
    value="PAB"
  >
    PAB
  </option>
  <option
    value="PEN"
  >
    PEN
  </option>
  <option
    value="PGK"
  >
    PGK
  </option>
  <option
    value="PHP"
  >
    PHP
  </option>
  <option
    value="PKR"
  >
    PKR
  </option>
  <option
    value="PLN"
  >
    PLN
  </option>
  <option
    value="PYG"
  >
    PYG
  </option>
  <option
    value="QAR"
  >
    QAR
  </option>
  <option
    value="RON"
  >
    RON
  </option>
  <option
    value="RSD"
  >
    RSD
  </option>
  <option
    value="RUB"
  >
    RUB
  </option>
  <option
    value="RWF"
  >
    RWF
  </option>
  <option
    value="SAR"
  >
    SAR
  </option>
  <option
    value="SBD"
  >
    SBD
  </option>
  <option
    value="SCR"
  >
    SCR
  </option>
  <option
    value="SDG"
  >
    SDG
  </option>
  <option
    value="SEK"
  >
    SEK
  </option>
  <option
    value="SGD"
  >
    SGD
  </option>
  <option
    value="SHP"
  >
    SHP
  </option>
  <option
    value="SKK"
  >
    SKK
  </option>
  <option
    value="SLL"
  >
    SLL
  </option>
  <option
    value="SOS"
  >
    SOS
  </option>
  <option
    value="SRD"
  >
    SRD
  </option>
  <option
    value="SSP"
  >
    SSP
  </option>
  <option
    value="STD"
  >
    STD
  </option>
  <option
    value="SVC"
  >
    SVC
  </option>
  <option
    value="SYP"
  >
    SYP
  </option>
  <option
    value="SZL"
  >
    SZL
  </option>
  <option
    value="THB"
  >
    THB
  </option>
  <option
    value="TJS"
  >
    TJS
  </option>
  <option
    value="TMT"
  >
    TMT
  </option>
  <option
    value="TND"
  >
    TND
  </option>
  <option
    value="TOP"
  >
    TOP
  </option>
  <option
    value="TRY"
  >
    TRY
  </option>
  <option
    value="TTD"
  >
    TTD
  </option>
  <option
    value="TWD"
  >
    TWD
  </option>
  <option
    value="TZS"
  >
    TZS
  </option>
  <option
    value="UAH"
  >
    UAH
  </option>
  <option
    value="UGX"
  >
    UGX
  </option>
  <option
    value="USD"
  >
    USD
  </option>
  <option
    value="UYU"
  >
    UYU
  </option>
  <option
    value="UZS"
  >
    UZS
  </option>
  <option
    value="VEF"
  >
    VEF
  </option>
  <option
    value="VND"
  >
    VND
  </option>
  <option
    value="VUV"
  >
    VUV
  </option>
  <option
    value="WST"
  >
    WST
  </option>
  <option
    value="XAF"
  >
    XAF
  </option>
  <option
    value="XAG"
  >
    XAG
  </option>
  <option
    value="XAU"
  >
    XAU
  </option>
  <option
    value="XBA"
  >
    XBA
  </option>
  <option
    value="XBB"
  >
    XBB
  </option>
  <option
    value="XBC"
  >
    XBC
  </option>
  <option
    value="XBD"
  >
    XBD
  </option>
  <option
    value="XCD"
  >
    XCD
  </option>
  <option
    value="XDR"
  >
    XDR
  </option>
  <option
    value="XOF"
  >
    XOF
  </option>
  <option
    value="XPD"
  >
    XPD
  </option>
  <option
    value="XPF"
  >
    XPF
  </option>
  <option
    value="XPT"
  >
    XPT
  </option>
  <option
    value="XTS"
  >
    XTS
  </option>
  <option
    value="YER"
  >
    YER
  </option>
  <option
    value="ZAR"
  >
    ZAR
  </option>
  <option
    value="ZMK"
  >
    ZMK
  </option>
  <option
    value="ZMW"
  >
    ZMW
  </option>
  <option
    value="BTC"
  >
    BTC
  </option>
  <option
    value="JEP"
  >
    JEP
  </option>
  <option
    value="GGP"
  >
    GGP
  </option>
  <option
    value="IMP"
  >
    IMP
  </option>
  <option
    value="XFU"
  >
    XFU
  </option>
  <option
    value="GBX"
  >
    GBX
  </option>
  <option
    value="CNH"
  >
    CNH
  </option>
  <option
    value="EEK"
  >
    EEK
  </option>
  <option
    value="GHS"
  >
    GHS
  </option>
  <option
    value="MTL"
  >
    MTL
  </option>
  <option
    value="TMM"
  >
    TMM
  </option>
  <option
    value="JPY"
  >
    JPY
  </option>
  <option
    value="ZWD"
  >
    ZWD
  </option>
  <option
    value="ZWL"
  >
    ZWL
  </option>
  <option
    value="ZWN"
  >
    ZWN
  </option>
  <option
    value="ZWR"
  >
    ZWR
  </option>
</select>
`;

exports[`<GiftCardCurrencySelector/> 1`] = `
<select
  defaultValue=""
  name="currencyCode"
  required={true}
>
  <option
    value=""
  >
    Select an option
  </option>
  <option
    value="AED"
  >
    AED
  </option>
  <option
    value="AFN"
  >
    AFN
  </option>
  <option
    value="ALL"
  >
    ALL
  </option>
  <option
    value="AMD"
  >
    AMD
  </option>
  <option
    value="ANG"
  >
    ANG
  </option>
  <option
    value="AOA"
  >
    AOA
  </option>
  <option
    value="ARS"
  >
    ARS
  </option>
  <option
    value="AUD"
  >
    AUD
  </option>
  <option
    value="AWG"
  >
    AWG
  </option>
  <option
    value="AZN"
  >
    AZN
  </option>
  <option
    value="BAM"
  >
    BAM
  </option>
  <option
    value="BBD"
  >
    BBD
  </option>
  <option
    value="BDT"
  >
    BDT
  </option>
  <option
    value="BGN"
  >
    BGN
  </option>
  <option
    value="BHD"
  >
    BHD
  </option>
  <option
    value="BIF"
  >
    BIF
  </option>
  <option
    value="BMD"
  >
    BMD
  </option>
  <option
    value="BND"
  >
    BND
  </option>
  <option
    value="BOB"
  >
    BOB
  </option>
  <option
    value="BRL"
  >
    BRL
  </option>
  <option
    value="BSD"
  >
    BSD
  </option>
  <option
    value="BTN"
  >
    BTN
  </option>
  <option
    value="BWP"
  >
    BWP
  </option>
  <option
    value="BYN"
  >
    BYN
  </option>
  <option
    value="BYR"
  >
    BYR
  </option>
  <option
    value="BZD"
  >
    BZD
  </option>
  <option
    value="CAD"
  >
    CAD
  </option>
  <option
    value="CDF"
  >
    CDF
  </option>
  <option
    value="CHF"
  >
    CHF
  </option>
  <option
    value="CLF"
  >
    CLF
  </option>
  <option
    value="CLP"
  >
    CLP
  </option>
  <option
    value="CNY"
  >
    CNY
  </option>
  <option
    value="COP"
  >
    COP
  </option>
  <option
    value="CRC"
  >
    CRC
  </option>
  <option
    value="CUC"
  >
    CUC
  </option>
  <option
    value="CUP"
  >
    CUP
  </option>
  <option
    value="CVE"
  >
    CVE
  </option>
  <option
    value="CZK"
  >
    CZK
  </option>
  <option
    value="DJF"
  >
    DJF
  </option>
  <option
    value="DKK"
  >
    DKK
  </option>
  <option
    value="DOP"
  >
    DOP
  </option>
  <option
    value="DZD"
  >
    DZD
  </option>
  <option
    value="EGP"
  >
    EGP
  </option>
  <option
    value="ERN"
  >
    ERN
  </option>
  <option
    value="ETB"
  >
    ETB
  </option>
  <option
    value="EUR"
  >
    EUR
  </option>
  <option
    value="FJD"
  >
    FJD
  </option>
  <option
    value="FKP"
  >
    FKP
  </option>
  <option
    value="GBP"
  >
    GBP
  </option>
  <option
    value="GEL"
  >
    GEL
  </option>
  <option
    value="GHS"
  >
    GHS
  </option>
  <option
    value="GIP"
  >
    GIP
  </option>
  <option
    value="GMD"
  >
    GMD
  </option>
  <option
    value="GNF"
  >
    GNF
  </option>
  <option
    value="GTQ"
  >
    GTQ
  </option>
  <option
    value="GYD"
  >
    GYD
  </option>
  <option
    value="HKD"
  >
    HKD
  </option>
  <option
    value="HNL"
  >
    HNL
  </option>
  <option
    value="HRK"
  >
    HRK
  </option>
  <option
    value="HTG"
  >
    HTG
  </option>
  <option
    value="HUF"
  >
    HUF
  </option>
  <option
    value="IDR"
  >
    IDR
  </option>
  <option
    value="ILS"
  >
    ILS
  </option>
  <option
    value="INR"
  >
    INR
  </option>
  <option
    value="IQD"
  >
    IQD
  </option>
  <option
    value="IRR"
  >
    IRR
  </option>
  <option
    value="ISK"
  >
    ISK
  </option>
  <option
    value="JMD"
  >
    JMD
  </option>
  <option
    value="JOD"
  >
    JOD
  </option>
  <option
    value="JPY"
  >
    JPY
  </option>
  <option
    value="KES"
  >
    KES
  </option>
  <option
    value="KGS"
  >
    KGS
  </option>
  <option
    value="KHR"
  >
    KHR
  </option>
  <option
    value="KMF"
  >
    KMF
  </option>
  <option
    value="KPW"
  >
    KPW
  </option>
  <option
    value="KRW"
  >
    KRW
  </option>
  <option
    value="KWD"
  >
    KWD
  </option>
  <option
    value="KYD"
  >
    KYD
  </option>
  <option
    value="KZT"
  >
    KZT
  </option>
  <option
    value="LAK"
  >
    LAK
  </option>
  <option
    value="LBP"
  >
    LBP
  </option>
  <option
    value="LKR"
  >
    LKR
  </option>
  <option
    value="LRD"
  >
    LRD
  </option>
  <option
    value="LSL"
  >
    LSL
  </option>
  <option
    value="LTL"
  >
    LTL
  </option>
  <option
    value="LVL"
  >
    LVL
  </option>
  <option
    value="LYD"
  >
    LYD
  </option>
  <option
    value="MAD"
  >
    MAD
  </option>
  <option
    value="MDL"
  >
    MDL
  </option>
  <option
    value="MGA"
  >
    MGA
  </option>
  <option
    value="MKD"
  >
    MKD
  </option>
  <option
    value="MMK"
  >
    MMK
  </option>
  <option
    value="MNT"
  >
    MNT
  </option>
  <option
    value="MOP"
  >
    MOP
  </option>
  <option
    value="MRO"
  >
    MRO
  </option>
  <option
    value="MUR"
  >
    MUR
  </option>
  <option
    value="MVR"
  >
    MVR
  </option>
  <option
    value="MWK"
  >
    MWK
  </option>
  <option
    value="MXN"
  >
    MXN
  </option>
  <option
    value="MYR"
  >
    MYR
  </option>
  <option
    value="MZN"
  >
    MZN
  </option>
  <option
    value="NAD"
  >
    NAD
  </option>
  <option
    value="NGN"
  >
    NGN
  </option>
  <option
    value="NIO"
  >
    NIO
  </option>
  <option
    value="NOK"
  >
    NOK
  </option>
  <option
    value="NPR"
  >
    NPR
  </option>
  <option
    value="NZD"
  >
    NZD
  </option>
  <option
    value="OMR"
  >
    OMR
  </option>
  <option
    value="PAB"
  >
    PAB
  </option>
  <option
    value="PEN"
  >
    PEN
  </option>
  <option
    value="PGK"
  >
    PGK
  </option>
  <option
    value="PHP"
  >
    PHP
  </option>
  <option
    value="PKR"
  >
    PKR
  </option>
  <option
    value="PLN"
  >
    PLN
  </option>
  <option
    value="PYG"
  >
    PYG
  </option>
  <option
    value="QAR"
  >
    QAR
  </option>
  <option
    value="RON"
  >
    RON
  </option>
  <option
    value="RSD"
  >
    RSD
  </option>
  <option
    value="RUB"
  >
    RUB
  </option>
  <option
    value="RWF"
  >
    RWF
  </option>
  <option
    value="SAR"
  >
    SAR
  </option>
  <option
    value="SBD"
  >
    SBD
  </option>
  <option
    value="SCR"
  >
    SCR
  </option>
  <option
    value="SDG"
  >
    SDG
  </option>
  <option
    value="SEK"
  >
    SEK
  </option>
  <option
    value="SGD"
  >
    SGD
  </option>
  <option
    value="SHP"
  >
    SHP
  </option>
  <option
    value="SKK"
  >
    SKK
  </option>
  <option
    value="SLL"
  >
    SLL
  </option>
  <option
    value="SOS"
  >
    SOS
  </option>
  <option
    value="SRD"
  >
    SRD
  </option>
  <option
    value="SSP"
  >
    SSP
  </option>
  <option
    value="STD"
  >
    STD
  </option>
  <option
    value="SVC"
  >
    SVC
  </option>
  <option
    value="SYP"
  >
    SYP
  </option>
  <option
    value="SZL"
  >
    SZL
  </option>
  <option
    value="THB"
  >
    THB
  </option>
  <option
    value="TJS"
  >
    TJS
  </option>
  <option
    value="TMT"
  >
    TMT
  </option>
  <option
    value="TND"
  >
    TND
  </option>
  <option
    value="TOP"
  >
    TOP
  </option>
  <option
    value="TRY"
  >
    TRY
  </option>
  <option
    value="TTD"
  >
    TTD
  </option>
  <option
    value="TWD"
  >
    TWD
  </option>
  <option
    value="TZS"
  >
    TZS
  </option>
  <option
    value="UAH"
  >
    UAH
  </option>
  <option
    value="UGX"
  >
    UGX
  </option>
  <option
    value="USD"
  >
    USD
  </option>
  <option
    value="UYU"
  >
    UYU
  </option>
  <option
    value="UZS"
  >
    UZS
  </option>
  <option
    value="VEF"
  >
    VEF
  </option>
  <option
    value="VND"
  >
    VND
  </option>
  <option
    value="VUV"
  >
    VUV
  </option>
  <option
    value="WST"
  >
    WST
  </option>
  <option
    value="XAF"
  >
    XAF
  </option>
  <option
    value="XAG"
  >
    XAG
  </option>
  <option
    value="XAU"
  >
    XAU
  </option>
  <option
    value="XBA"
  >
    XBA
  </option>
  <option
    value="XBB"
  >
    XBB
  </option>
  <option
    value="XBC"
  >
    XBC
  </option>
  <option
    value="XBD"
  >
    XBD
  </option>
  <option
    value="XCD"
  >
    XCD
  </option>
  <option
    value="XDR"
  >
    XDR
  </option>
  <option
    value="XOF"
  >
    XOF
  </option>
  <option
    value="XPD"
  >
    XPD
  </option>
  <option
    value="XPF"
  >
    XPF
  </option>
  <option
    value="XPT"
  >
    XPT
  </option>
  <option
    value="XTS"
  >
    XTS
  </option>
  <option
    value="YER"
  >
    YER
  </option>
  <option
    value="ZAR"
  >
    ZAR
  </option>
  <option
    value="ZMK"
  >
    ZMK
  </option>
  <option
    value="ZMW"
  >
    ZMW
  </option>
  <option
    value="BTC"
  >
    BTC
  </option>
  <option
    value="JEP"
  >
    JEP
  </option>
  <option
    value="GGP"
  >
    GGP
  </option>
  <option
    value="IMP"
  >
    IMP
  </option>
  <option
    value="XFU"
  >
    XFU
  </option>
  <option
    value="GBX"
  >
    GBX
  </option>
  <option
    value="CNH"
  >
    CNH
  </option>
  <option
    value="EEK"
  >
    EEK
  </option>
  <option
    value="GHS"
  >
    GHS
  </option>
  <option
    value="MTL"
  >
    MTL
  </option>
  <option
    value="TMM"
  >
    TMM
  </option>
  <option
    value="JPY"
  >
    JPY
  </option>
  <option
    value="ZWD"
  >
    ZWD
  </option>
  <option
    value="ZWL"
  >
    ZWL
  </option>
  <option
    value="ZWN"
  >
    ZWN
  </option>
  <option
    value="ZWR"
  >
    ZWR
  </option>
</select>
`;
